core/src/main/scala/kafka/server/KafkaServerStartable.scala

```diff
@@ -37,10 +37,10 @@ class KafkaServerStartable(val serverConfig: KafkaConfig, reporters: Seq[KafkaMe
   def startup() {
     try server.startup()
     catch {
-      case _: Throwable =>
+      case e: Throwable =>
         // KafkaServer.startup() calls shutdown() in case of exceptions, so we invoke `exit` to set the status code
         fatal("Exiting Kafka.")
-        Exit.exit(1)
+        Exit.exit(1, Some(s"Exiting Kafka due to exception during startup: $e"))
```

Comment:
Because we normally log because calling `exit` (like we did above), I was thinking we could just add some logging to `exit` and `halt` in `org.apache.kafka.common.utils.Exit`. It would have a generic message saying that `exit` or `halt` was invoked by Kafka so that it would be easy to grep for.

Reply:
We don't call `exit` or `halt` from the signal handler, so do you mean we will store some state and print the appropriate message? Or do you mean another message in `Exit` with a generic message in addition to the message from signal handler?

Reply:
@ijuma Thanks for the review. Updated and rebased.